flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpReporter.java

```diff
@@ -67,20 +67,20 @@ public DatadogHttpReporter(
             String proxyHost,
             int proxyPort,
             int maxMetricsPerRequestValue,
-            DataCenter dataCenter,
+            String dataCenterUrl,
             String tags,
             boolean useLogicalIdentifier) {
         this.maxMetricsPerRequestValue = maxMetricsPerRequestValue;
         this.useLogicalIdentifier = useLogicalIdentifier;
-        this.client = new DatadogHttpClient(apiKey, proxyHost, proxyPort, dataCenter, true);
+        this.client = new DatadogHttpClient(apiKey, proxyHost, proxyPort, dataCenterUrl, true);
         this.configTags = getTagsFromConfig(tags);
 
         LOGGER.info(
-                "Configured DatadogHttpReporter with {tags={}, proxyHost={}, proxyPort={}, dataCenter={}, maxMetricsPerRequest={}",
+                "Configured DatadogHttpReporter with {tags={}, proxyHost={}, proxyPort={}, dataCenterUrl={}, maxMetricsPerRequest={}",
```

Comment:
I was thinking you would keep the original constructor and deprecate it and add a new constructor for the url.
The log here would then come out reflecting the users config. 

I see you have refactored the callers to send through the url. deprecating might be cleaner. WDYT ?

The methods should all have @publicEvolving @internal tags as appropriate  in all the methods in all the classes